### PR TITLE
fix iOS build, remove deprecated API

### PR DESCRIPTION
**Pull request by nmn**
I saw that the tests were failing on iOS, so I made the changes necessary to make the tests pass on iOS and iPadOS in addition to macOS.

### CODE REVIEW

File: Sources/Syntax/Parser.swift [hunk 1]
```diff
@@ -112,7 +112,7 @@ extension Parser {
         }
 
         let storage: MemoizationStorage
-        if #available(OSX 10.15, *) {
+        if #available(OSX 10.15, iOS 13, *) {
```
Comment (nerdsupremacist) [inline Sources/Syntax/Parser.swift]: nit: could we add support for tvOS and watchOS here too?  Pretty sure it's: tvOS 13 and watchOS 6

**Review by nerdsupremacist — APPROVED** [review]
LGTM

thanks for spotting this!

**Comment by nerdsupremacist** [comment]
no idea why the linux build is failing... it appears to be fine

**Comment by nmn** [comment]
I tried to make the tests pass on even more platforms. But I'm assuming linux will still fail.

I also don't understand why, after I install `Syntax` as a dependency for a Swift Package, it can't be found when I import it. Some package configuration thing is probably still broken.

**Pull request merged**